pyomo/core/base/boolean_var.py

```diff
@@ -0,0 +1,590 @@
+        value       The numeric value of this variable.
+    """
+    __slots__ = ()
+
+    def __init__(self, component=None):
+        self._component = weakref_ref(component) if (component is not None) \
+                          else None
+
+    def is_fixed(self):
+        """Returns True if this variable is fixed, otherwise returns False."""
+        return self.fixed
+
+    def is_constant(self):
+        """Returns False because this is not a constant in an expression."""
+        return False
+
+    def is_variable_type(self):
+        """Returns True because this is a variable."""
+        return True
+
+    def is_potentially_variable(self):
+        """Returns True because this is a variable."""
+        return True
+
+    def set_value(self, val, valid=False):
+        """
+        Set the value of this numeric object, after
+        validating its value. If the 'valid' flag is True,
+        then the validation step is skipped.
+        """
+        if valid or self._valid_value(val):
+            self.value = val
+            self.stale = False
+
+    def _valid_value(self, val, use_exception=True):
+        """
+        Validate the value.  If use_exception is True, then raise an
+        exception.
+        """
+        ans = val is None or val in (True, False)
+        if not ans and use_exception:
+            raise ValueError(
+                "Logical value `%s` (%s) is not True, False, or None"
+                % (val, type(val)))
+        return ans
+
+    def clear(self):
+        self.value = None
+
+
+    def __call__(self, exception=True):
+        """Compute the value of this variable."""
+        return self.value
+
+    @property
+    def value(self):
+        """Return the value for this variable."""
+        raise NotImplementedError
+
+    @property
+    def domain(self):
+        """Return the domain for this variable."""
+        raise NotImplementedError
+
+    @property
+    def fixed(self):
+        """Return the fixed indicator for this variable."""
+        raise NotImplementedError
+
+    @property
+    def stale(self):
+        """Return the stale indicator for this variable."""
+        raise NotImplementedError
+
+    def fix(self, *val):
+        """
+        Set the fixed indicator to True. Value argument is optional,
+        indicating the variable should be fixed at its current value.
+        """
+        raise NotImplementedError
+
+    def unfix(self):
+        """Sets the fixed indicator to False."""
+        raise NotImplementedError
+
+    free=unfix
+
+    def to_string(self, verbose=None, labeler=None, smap=None, compute_values=False):
+        """Return the component name"""
+        if self.fixed and compute_values:
+            try:
+                return str(self())
+            except:
+                pass
+        if smap:
+            return smap.getSymbol(self, labeler)
+        return self.name
+
+
+class _GeneralBooleanVarData(_BooleanVarData):
```

Comment:
This is a somewhat unresolved matter of design taste, I think. @jsiirola any opinion here?

Reply:
I thought the majority of transformations added information onto the model to manage these kinds of mappings. This feels like the transformation from logic to mip is leaking into the component. Having said this, I don't think it should hold up the PR, because I don't really this has a significant public facing component. If we later moved this kind of association to data on the model, I don't think the API would change. Agreed?

Reply:
Let me know if you have any other thoughts. I am happy to mark this as resolved - just would like to capture the design discussion.

Reply:
I added an issue to make sure we discuss this before the next release.